### DISCUSSION TITLE
What is the difference between transaction cost and execution cost?

### DISCUSSION THREAD

**Discussion by dalteko** [Cyfrin/foundry-full-course-cu#119] [General]
Hey there, finishing up Lesson 5 of this course and making a discussion post question. 

Not using AI for this as it's an example, but will find the answer if nobody responds. 

Question: 
![Screenshot 2023-06-20 at 7 42 32 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/104739909/a2adefe8-d427-42a0-9aff-aabf698c5190)

When I deploy the finished FundMe{} contract, it has three types of gas costs: 
-gas 
-transaction cost
-execution cost

In Lesson 4, Patrick refers to the transaction cost as the cost of deploying the contract, which makes sense to me. But what is the execution cost in this context? And what is the overarching difference between all three when you're viewing a public variable from a contract, calling a function, and deploying a contract?

**Comment by alymurtazamemon** [comment]
@dalteko 

1. `gas` is the unit of measure just like the Patrol for cars, we say it consumes 3 Liters for this ride, but this is not the cost because the cost will the price of 3 Liters of Patrol.
2. `transaction cost` as its name says is the overall cost of a transaction from initiating to all other operations such as input/output handling, contract creation, executing contract logic, updating storage variables, interacting with external contracts, etc.
3. `execution cost` is the sub-part of the `transaction cost`, it only tells the price of running the operations/logic inside the contract.

**Reply by dalteko** [reply]
thank you! this makes sense
